substrate/frame/contracts/src/wasm/prepare.rs

```diff
@@ -263,17 +268,25 @@ where
 					Determinism::Enforced,
 					stack_limits,
 					LoadingMode::Checked,
+					CompilationMode::Eager,
 				) {
 					*determinism = Determinism::Enforced;
 					module
 				} else {
-					LoadedModule::new::<T>(code, Determinism::Relaxed, None, LoadingMode::Checked)?
+					LoadedModule::new::<T>(
+						code,
+						Determinism::Relaxed,
+						None,
+						LoadingMode::Checked,
+						CompilationMode::Eager,
+					)?
 				},
 			Determinism::Enforced => LoadedModule::new::<T>(
 				code,
 				Determinism::Enforced,
 				stack_limits,
 				LoadingMode::Checked,
+				CompilationMode::Eager,
```

Comment:
What is the reasoning to compile eagerly when validating? Is it possible for lazy compilation to fail on a validated (but not fully compiled) module? @Robbepop

Reply:
When using lazy compilation we also defer Wasmi compilation errors e.g. when exceeding specific Wasmi compilation limits.

Reply:
Okay then we have to check eagerly.